### PR TITLE
DSD-834: consistent naming for props, enums, files

### PR DESCRIPTION
**Pull request by EdwinGuzman**
Fixes JIRA ticket [DSD-834](https://jira.nypl.org/browse/DSD-834)

## This PR does the following:

- This is still a work in progress because we need to talk about the `Hero` and its image-related props, but take a look at the changelog for a full description of all the changes:

```
### Adds

- Created new `LayoutTypes` enum for row and column layouts.

### Removals

- Removes the `CardLayouts`, `CheckboxGroupLayoutTypes`, `RadioGroupLayoutTypes`, and `SkeletonLoaderLayouts` enums.
- Removes the `CardTypes.tsx`,` CheckboxGroupLayoutTypes.tsx`, and `RadioGroupLayoutTypes.tsx` files.

### Updates

- Renames the `Accordion` prop `contentData` to `accordionData`.
- Renames the `Breadcrumbs` prop `colorVariant` to `breadcrumbsType`.
- Renames the `Card` prop `center` to `isCentered`.
- Updates the `Card` prop `layout` to use the `LayoutTypes` enum instead of `CardLayouts`.
- Updates the `CheckboxGroup` prop `layout` to use the `LayoutTypes` enum instead of `CheckboxGroupLayoutTypes`.
- Renames the `Heading` prop `displaySize` to `size.
- Renames the `HeadingDisplaySizes` enum to `HeadingSizes.
- Renames the `Hero` prop `image` to `imageComponent`.
- Renames the `Image` prop `imageSize` to `size`.
- Renames the `Notification` prop `centered` to `isCentered`.
- Updates the `RadioGroup` prop `layout` to use the `LayoutTypes` enum instead of `RadioGroupLayoutTypes`.
- Internal updates to the `SearchBar` component based on updates from the `Select` and `TextInput` components.
- Renames the `Select` prop `type` to `selectType`.
- Updates the `SkeletonLoader` prop `layout` to use the `LayoutTypes` enum instead of `SkeletonLoaderLayouts`.
- Renames the `StructuredContentImage` prop `imageSize` to `size` (this component is based on the `Image` component).
- Internal updates to the `StructuredContent` component based on updates from the `Image` and `Heading` components.
- Renames the `Tabs` prop `contentData` to `tabsData`.
- Renames the `Text` prop `displaySize` to `size`.
- Renames the `TextInput` prop `variant` to `textInputType`.
- Renames the `ToggleSizes.tsx` file to `ToggleTypes.tsx`.
```

It is worth noting that for the `size` prop, even though it's the same name, we cannot use the same enum value. Every component has a slightly different enum object to use for its size:

| component | values |
| ----------- | ------ |
| Toggle       |  Large, Small |
| text | Default, Caption, Tag, Mini |
| progressindicator |  Default, Small |
| logo | ExtraExtraSmall, ExtraSmall, Medium, Large, Default
| icon | Small, Medium, Large, ExtraLarge, ExtraExtraLarge, ExtraExtraExtraLarge
| image | Default, ExtraExtraSmall, ExtraSmall, Small,  Medium, Large
| heading | Primary, Secondary, Tertiary, Callout 

## How has this been tested?

Storybook and tests. No functional updates just naming convention updates.

## Accessibility concerns or updates

- N/A

### Checklist:

- [ ] I have updated the Storybook documentation accordingly.
- [ ] I have added relevant accessibility documentation for this pull request.
- [ ] All new and existing tests passed.

### Front End Review:

- [ ] View [the example in Storybook]()

### CODE REVIEW

**Comment by EdwinGuzman** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr892-tfew8tb9ymfbizvyvlanhovaj4toozuc.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/622683f501c6d20b85350e27

**Comment by bigfishdesign13** [comment]
@EdwinGuzman 
Let’s try to bring some more consistency to the `size` enum values.
- `Toggle`: change `Large` and `Default`
- `Icon`: add `Default` and have it act the same as `Image` and `Logo`
- `Heading`: should we add `Default` and have it align with the `Secondary` styles?
- Anything that has `ExtraExtra` seems foolish, but can we recalibrate and come up with a better scale?

**Comment by EdwinGuzman** [comment]
I updated the `Toggle` to have `Default` and `Small`. The `Icon` component is also updated but they don't all act the same. The default for `Logo` is `Medium` and maybe it should stay that way? Or else the default is 100% width which is huge.

`Heading` has its own logic which includes different h* sizes, so that one may not work or should be refactored.

Yes, but let's make the scaling of enum values a different ticket.

**Comment by EdwinGuzman** [comment]
Marking this as ready to review and latest round of updates:
```
### Adds
- Adds the value "Default" to the `IconSizes` enum.

### Updates
- Updates the `Card` image-related props into one main prop named `imageProps`. This new prop contains the following properties: alt, aspectRatio, caption, component, credit, isAtEnd, size, and src.
- Updates the `Hero` prop `image` to `imageAlt` and `imageSrc`. Internally, an `Image` component is created.
- Renames the `Image` component props `imageAspectRatio` to `aspectRatio`, `imageCaption` to `caption`, and `imageCredit` to `credit`.
- Updates the `StructuredContent` image-related props into one main prop named `imageProps`. This new prop contains the following properties: alt, aspectRatio, caption, component, credit, position, size, and src.
- Renames the `ToggleSizes.tsx` file to `ToggleTypes.tsx`. Updates the values from `Large` and `Small` to `Default` and `Small`.
- Minor update to the logic for the `ProgressIndicator` sizing prop and styles.
```

Note: In Storybook for `Card` and `StructuredContent`, you'll now see and be able to update `imageProps.alt`, for example, instead of `imageAlt`. It looks odd but I think it's clear that that `imageProps` is an object and `alt` is the property for it that the Controls is updating/modifying. An extra step must be taken in the Story's args though but it's pretty clear, I think, from the Story file.

**Comment by EdwinGuzman** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr892-tfew8tb9ymfbizvyvlanhovaj4toozuc.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/622683f501c6d20b85350e27

File: src/components/Card/Card.tsx
Comment (bigfishdesign13) [inline src/components/Card/Card.tsx:30]: Can an interface be setup as a global reusable object?

File: src/components/Hero/Hero.tsx
Comment (bigfishdesign13) [inline src/components/Hero/Hero.tsx]: Do you think we should be more explicit with the labelling on the warning messages?
```
NYPL Reservoir Hero Component: …
```
Doing so would impact a lot of other components as well.

File: src/components/Hero/Hero.tsx
Comment (bigfishdesign13) [inline src/components/Hero/Hero.tsx:86]: I think we should also have a test to make sure when one of these props is set the other is also set.

File: src/theme/components/icon.ts
Comment (bigfishdesign13) [inline src/theme/components/icon.ts:43]: The `default` size should go to 100% width.  That is how `Image` and `Logo` treat `default`.

File: src/components/StructuredContent/StructuredContent.stories.mdx
Comment (bigfishdesign13) [inline src/components/StructuredContent/StructuredContent.stories.mdx:129]: Why are the `imageProps` props in quotes?

**Review by bigfishdesign13 — COMMENTED** [review]
Phew!  Wow, that was quite a PR.  I have a few questions sprinkled throughout.

One thing that I would like to see is more consistency with how the sections and info are handled on the stories for `Image`, `Icon` and `Logo`.  For example, the `Size` section on each story is a bit different.  Do you want to attack those changes in this PR or setup a new ticket for that?

File: src/components/Card/Card.tsx
Comment (EdwinGuzman) [reply src/components/Card/Card.tsx:30]: Kind of. I'm pulling in `ImageProps` for the `CardImage` component. Maybe the naming could be slightly better.

The other place something like this happens in the `StructuredContent` but the only difference is this has `isAtEnd` and `StructuredContent` has `position`. Otherwise, I can update this.

File: src/components/Hero/Hero.tsx
Comment (EdwinGuzman) [reply src/components/Hero/Hero.tsx]: I think that's too long. Maybe `NYPL Hero: ...` ? Yeah, this may need another ticket for consistency everywhere.

File: src/components/StructuredContent/StructuredContent.stories.mdx
Comment (EdwinGuzman) [reply src/components/StructuredContent/StructuredContent.stories.mdx:129]: It's in quotes so it appears in Storybook's args table and so that the Controls can pick up the changes. Likewise, when getting this value from the `args` object, you have to do `args["imageProps.alt"]` -- which is pretty dumb. I expected `args.imageProps.alt` but this doesn't work in Storybook and others also submitted issues regarding nested values. The default way storybook does this isn't very intuitive so by making it `"imageProps.alt"`, it's hopefully clear that you are updating the `alt` property of the `imageProps` prop object.

Comment (bigfishdesign13) [reply src/components/StructuredContent/StructuredContent.stories.mdx:129]: Got it.  Thanks.  And yes, I thought what was showing in the ArgsTable was clear.

File: src/theme/components/icon.ts
Comment (bigfishdesign13) [reply src/theme/components/icon.ts:43]: Sorry, I forgot I had this in here.  This is obviously a requested change.  Does the thumbs-up mean you will make that change in this PR?

**Comment by bigfishdesign13** [comment]
> `Heading` has its own logic which includes different h* sizes, so that one may not work or should be refactored.

Right, this is a tricky one.  I think it is hard to identify “default" styles for a heading, so let’s leave that one alone, at least for now.

File: src/components/Hero/Hero.stories.mdx
Comment (oliviawongnyc) [inline src/components/Hero/Hero.stories.mdx:41]: Would it make sense to use one prop, `imageProps`,  here like you did in the Card component?

File: src/components/Image/Image.stories.mdx
Comment (oliviawongnyc) [inline src/components/Image/Image.stories.mdx:107]: Wondering why we removed the `image` prefix from all but type?

File: src/theme/components/icon.ts
Comment (EdwinGuzman) [reply src/theme/components/icon.ts:43]: Yes, that I'll make a change. For this one, this is width 100%:
Screen Shot 2022-03-10 at 12 55 51 PM

and this is width and height 100%:
Screen Shot 2022-03-10 at 12 55 42 PM

It didn't seem right to me so that's why I made it similar to `medium`. Which one should it be?

File: src/components/Hero/Hero.stories.mdx
Comment (EdwinGuzman) [reply src/components/Hero/Hero.stories.mdx:41]: Yep! But this is going to be another ticket/PR because the `Hero` component needs to be refactored: https://jira.nypl.org/browse/DSD-614

File: src/components/Image/Image.stories.mdx
Comment (EdwinGuzman) [reply src/components/Image/Image.stories.mdx:107]: Ah, because this `imageType` is related to the Chakra "variant" for its style. We were thinking of using `type` but the problem is that `type` is an actual HTML attribute and that will be confusing. For example, the HTML button element can take a `type` attribute and it can be set to "submit", "button", or "reset" `<button type="submit">Submit</button>`. To be consistent, the prop to change the variant is `buttonType`, while there also is a separate `type` prop for "submit", "button", and "reset" `<Button buttonType={ButtonTypes.Secondary} type="submit">Submit</Button>`.

**Review by oliviawongnyc — APPROVED** [review]
Wow! A lot of work here. Looks like a big and thoughtful improvement. Excited to implement in the test app.

File: src/components/Hero/Hero.tsx
Comment (EdwinGuzman) [reply src/components/Hero/Hero.tsx]: https://jira.nypl.org/browse/DSD-854

File: src/theme/components/icon.ts
Comment (EdwinGuzman) [reply src/theme/components/icon.ts:43]: I'm making it `width: "auto"`. It is very interesting that it's so similar to `Logo` but that one has `maxWidth` and `Icon` has `width`. Switching to maxWidth doesn't seem to work though....

**Comment by EdwinGuzman** [comment]
@bigfishdesign13 updated the width of the `Icon`.

**Comment by EdwinGuzman** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr892-tfew8tb9ymfbizvyvlanhovaj4toozuc.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/622683f501c6d20b85350e27

**Comment by EdwinGuzman** [comment]
@bigfishdesign13 the last update looks good but can you add for the `Icon` stories that you used a specific size for each section? For example, in the Colors section, it'd be good to point out that the size for the color examples are all `IconSizes.ExtraExtraLarge`.

**Comment by bigfishdesign13** [comment]
> @bigfishdesign13 the last update looks good but can you add for the `Icon` stories that you used a specific size for each section? For example, in the Colors section, it'd be good to point out that the size for the color examples are all `IconSizes.ExtraExtraLarge`.

@EdwinGuzman Thanks for the suggestions.  Please see the updates for `Icon` and `Logo`.

**Comment by bigfishdesign13** [comment]
@EdwinGuzman The `conflicts` are causing the Tugboat build to fail, right?

**Comment by EdwinGuzman** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr892-tfew8tb9ymfbizvyvlanhovaj4toozuc.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/622683f501c6d20b85350e27

**Comment by EdwinGuzman** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr892-tfew8tb9ymfbizvyvlanhovaj4toozuc.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/622683f501c6d20b85350e27

**Comment by EdwinGuzman** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr892-tfew8tb9ymfbizvyvlanhovaj4toozuc.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/622683f501c6d20b85350e27

**Comment by bigfishdesign13** [comment]
@EdwinGuzman This looks like it is ready to go.  Is there anything else you wanted to do before merging?

**Comment by EdwinGuzman** [comment]
Not sure. I'll update and merge!

**Comment by EdwinGuzman** [comment]
I'm going to wait until @oliviawongnyc 's latest PR on the Accordion is merged in.

**Comment by EdwinGuzman** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr892-tfew8tb9ymfbizvyvlanhovaj4toozuc.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/622683f501c6d20b85350e27

**Comment by EdwinGuzman** [comment]
Going to merge this in.

**Pull request merged**
